pandas/core/indexes/multi.py

```diff
@@ -3052,13 +3054,19 @@ def _update_indexer(idxr, indexer=indexer):
             elif is_list_like(k):
                 # a collection of labels to include from this level (these
                 # are or'd)
+                # Find out if the list_like label are sorted as the levels or not
+                if not need_sort:
+                    k_codes = self.levels[i].get_indexer(k)
+                    k_codes = k_codes[k_codes >= 0]  # Filter absent keys
+                    need_sort = not (k_codes[:-1] < k_codes[1:]).all()
```

Comment:
I don't recall: does NumPy short-circut `any`? If so it may be faster to change the comparision to `>=`, the `all` to an `any` and remove the `not` (haven't tested).

Reply:
According to this discussion  https://stackoverflow.com/questions/45771554/why-numpy-any-has-no-short-circuit-mechanism#45773662, neither `all` nor `any` short-circuit anymore.
I will changed it however, as it approve readability.
Also, it point out an error I made, it should have been `<=` here, as two consecutive equals elements are still sorted. Nice catch.